tests/lax_numpy_test.py

```diff
@@ -3873,14 +3873,21 @@ def testAstypeBool(self, from_dtype, use_method, to_dtype='bool'):
   @jtu.sample_product(
     change_dtype=[True, False],
     copy=[True, False],
+    change_device=[True, False],
   )
-  def testAstypeCopy(self, change_dtype, copy):
+  @jtu.run_on_devices("gpu")
+  def testAstypeCopy(self, change_dtype, copy, change_device):
```

Comment:
Please write more exhaustive tests covering different arrays types and different sharding/devices.